src/core/ext/filters/client_channel/resolver/polling_resolver.cc

```diff
@@ -58,7 +58,8 @@ PollingResolver::PollingResolver(ResolverArgs args,
       tracer_(tracer),
       interested_parties_(args.pollset_set),
       min_time_between_resolutions_(min_time_between_resolutions),
-      backoff_(backoff_options) {
+      backoff_(backoff_options),
+      event_engine_(channel_args.GetObjectRef<EventEngine>()) {
```

Comment:
I should have noticed this in my last pass through: we don't need to store another shared_ptr to the engine since the shared_ptr in channel_args keeps the engine alive, a raw pointer will suffice. This can be `channel_args_.GetObject<EventEngine>()`.

Reply:
(please comment that sharing!)